src/main/java/org/sagebionetworks/bridge/services/AssessmentService.java

```diff
@@ -0,0 +1,454 @@
+        // references to a collection." I think this could be fixed by using transaction annotations 
+        // that could be put on service methods so they can call multiple DAO calls in a session, 
+        // but we'd also have to change HibernateHelper to use the existing transaction rather
+        // than opening a new transaction.
+        if (original.getTags() != null) {
+            assessmentToPublish.setTags(ImmutableSet.copyOf(original.getTags()));    
+        }
+        original.setOriginGuid(assessmentToPublish.getGuid());
+        
+        return dao.publishAssessment(appId, original, assessmentToPublish);
+    }
+    
+    /**
+     * Takes an assessment in the shared context and clones it into the caller's local context, using the 
+     * specified organization as the new owner of the local assessment (must be an organization the caller
+     * is associated to). If the identifier already exists in this study, the revision number is adjusted
+     * appropriately. The origin GUID is set to the GUID of the shared assessment.
+     */
+    public Assessment importAssessment(String appId, String ownerId, String guid) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(guid));
+        
+        if (isBlank(ownerId)) {
+            throw new BadRequestException("ownerId parameter is required");
+        }
+        checkOwnership(appId, ownerId);
+
+        Assessment assessment = getAssessmentByGuid(SHARED_STUDY_ID_STRING, guid);
+        
+        // Figure out what revision this should be in the new app context if the identifier already exists
+        int revision = nextRevisionNumber(appId, assessment.getIdentifier());
+        assessment.setRevision(revision);
+        assessment.setOriginGuid(assessment.getGuid());
+        assessment.setOwnerId(ownerId);
+        
+        return createAssessmentInternal(appId, assessment);
+    }
+        
+    public void deleteAssessment(String appId, String guid) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(guid));
+        
+        Assessment assessment = getAssessmentByGuid(appId, guid);
+        if (assessment.isDeleted()) {
+            throw new EntityNotFoundException(Assessment.class);
+        }
+        checkOwnership(appId, assessment.getOwnerId());
+        
+        assessment.setDeleted(true);
+        assessment.setModifiedOn(getModifiedOn());
+        dao.saveAssessment(appId, assessment);
+    }
+        
+    public void deleteAssessmentPermanently(String appId, String guid) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(guid));
+        
+        Optional<Assessment> opt = dao.getAssessment(appId, guid);
+        if (opt.isPresent()) {
+            dao.deleteAssessment(appId, opt.get());    
+        }
+    }
+    
+    private Tuple<String> parseOwnerId(String guid, String ownerId) {
```

Comment:
Is it worth making this package-scoped so you can unit test the different branches in here?

Reply:
Did this and added three tests (realized I can prevent splitting on all but the first colon so it's a little stronger now).